fables/parse.py

```diff
@@ -30,17 +31,36 @@
 FRACTION_OF_BLANK_HEADERS_ALLOWED = 0.5
 
 
-def sniff_delimiter(bytesio: IO[bytes]) -> str:
-    # TODO(Thomas: 03/06/2019): Assuming utf-8 for now. Later should consider
-    #                           use of chardet and/or cchardet to auto-detect
-    #                           encoding.
-    sample = bytesio.read(1024 * 4).decode(encoding="utf-8")
+def sniff_delimiter(bytesio: IO[bytes], encoding: str) -> str:
+    if encoding is not None:
+        sample = bytesio.read(1024 * 4).decode(encoding=encoding)
+    else:
+        sample = bytesio.read(1024 * 4).decode()
```

Comment:
[non-blocking] If `encoding` is `None`, do we want to stick with the previous behavior of defaulting to `utf-8`? If so, this if-else can be simplified a bit by doing `encoding = encoding if encoding is not None else "utf-8"`, followed by `bytesio.read...`. Otherwise, this would default to the system default encoding, which is fine, but is just a slight change in behavior from before.